gui/pregnancy.lua

```diff
@@ -0,0 +1,328 @@
+
+function PregnancyGui:findName(unit)
+    if unit.name.has_name then
+        return dfhack.TranslateName(unit.name)
+    elseif unit.name.nickname ~= "" then
+        return unit.name.nickname
+    else return ('Unnamed %s. (Unit id:%s)'):format(
+        string.upper(df.global.world.raws.creatures.all[unit.race].name[0]),
+        unit.id
+    )
+    end
+end
+
+function PregnancyGui:findSpouse(unit)
+    local historical_spouse, spouse_loc, spouse, spouseid
+    local culled = false
+
+    --setting variables for if mother or father are local, followed by finding the father's spouse if he is not local
+    if self.father == unit or self.mother == unit then
+        spouseid = unit.relationship_ids.Spouse
+        spouse = df.unit.find(spouseid)
+    elseif self.father_historical == unit then
+        for index, relation in pairs(unit.histfig_links) do
+            if relation._type == df.histfig_hf_link_spousest then
+                historical_spouse=df.historical_figure.find(relation.target_hf)
+                if not historical_spouse then culled = true --there was an id, but there wasn't a histfig with that id (due culling)
+                elseif df.global.plotinfo.site_id==historical_spouse.info.whereabouts.site then
+                    spouse_loc = 'local'
+                else spouse_loc = 'offsite'
+                end
+            end
+        end
+        return {dfhack.TranslateName(historical_spouse.name),spouse_loc,historical_spouse}
+    end
+
+    --if the spouse is local this should identify them:
+    if spouse then
+        historical_spouse = df.historical_figure.find(spouse.hist_figure_id) or false
+        spouse_loc = 'local'
+    end
+
+    --if spouse is not local (offsite):
+    if spouseid > -1 and not spouse then --spouse exists but isnt on the map, so search historical units:
+        local historical_unit = df.historical_figure.find(unit.hist_figure_id)
+        for index, relation in pairs(historical_unit.histfig_links) do
+            if relation._type == df.histfig_hf_link_spousest then
+                historical_spouse=df.historical_figure.find(relation.target_hf)
+                if not historical_spouse then culled = true --there was an id, but there wasn't a histfig with that id (due culling)
+                elseif df.global.plotinfo.site_id==historical_spouse.info.whereabouts.site then--i dont think this should ever be true
+                    spouse_loc = 'local'
+                else spouse_loc = 'offsite'
+                end
+            end
+        end
+    end
+    if culled then 
+        return {'Unknown','culled'}
+    else 
+        return {dfhack.TranslateName(historical_spouse.name),spouse_loc,historical_spouse}
+    end
+end
+
+function PregnancyGui:CreatePregnancy()
+	local genes,father_id,father_caste,father_name
+    local bypass = true
+    local force = self.subviews.Force:getOptionValue()
+
+    local count = #self.msg
+    for i=0, count do self.msg[i]=nil end --empty self.msg
```

Comment:
```suggestion
    self.msg = {}
```

Reply:
Hmm... i tried this but only seem to get empty strings in the label widgets...